code/controllers/subsystems/supply.dm

```diff
@@ -264,16 +285,61 @@ SUBSYSTEM_DEF(supply)
 	var/reason = null
 	var/orderedrank = null //used for supply console printing
 
-/datum/controller/subsystem/supply/proc/find_item_value(var/obj/object) //here we get the value of the items being traded
+/datum/controller/subsystem/supply/proc/make_trade(var/obj/object, var/count = 1)
+	. = find_item_value(object, count)
+	if(.)
+		sold_items[object.type] += count
+	return .
+
+/datum/controller/subsystem/supply/proc/find_item_value(var/obj/object, var/count = 1, var/use_reinf_material = FALSE) //here we get the value of the items being traded
 	if(!object)
 		return 0
 
-	//this uses the default SS13 item_worth procs so its a good fallback
-	. = get_value(object)
+	var/sell_modifier = 1
 
-	var/datum/trade_item/T
+	if(GLOB.using_map.using_new_cargo)
+		if(istype(object, /obj/item/stack/material))	//Materials aren't effected by reduced value/compounded prices. Apply any needed modifiers and sell as-is
+			var/obj/item/stack/material/M = object
+			sell_modifier = GLOB.using_map.new_cargo_inflation * 0.25
+			if(use_reinf_material)
+				return round(round(count * 0.5) * M.reinf_material.sale_price * sell_modifier)
+			return round(count * M.material.sale_price * sell_modifier)
+		
+		else if(GLOB.using_map.trading_faction?.reputation > 50)
+			sell_modifier = (((GLOB.using_map.trading_faction.reputation - 50) / 100) / 2) + 1	//0.5% ~ 25% bonus
+
+	else
+		if(istype(object, /obj/item/stack/material))
```

Comment:
this is already handled on line 152, is it not?

Reply:
L152 calls `find_item_value()` for each material present in a stack and adds the resulting value to the list, it doesn't actually handle how much each stack/material is worth. The idea was that if we wanted to change how materials are sold/calculated in the future, all of that is under the `find_item_value` proc and L152 just handles the splitting (like if we wanted to have compounded sales effect materials too later down the road)